Initial release

iyourshaw opened the pull request:
This is the initial release of this project.  It introduces a complete set of Java classes to be used for serializing and deserializing standard XER and JER encoded ASN.1 for the SAE J2735/2024 specification, with a comprehensive set of unit tests and code coverage metrics, and a command line tool for generating example messages.

The code was reviewed in detail the following PRs:

* [J2735-2024 Part 1: Runtime library and structure](https://github.com/CDOT-CV/jpo-asn-pojos/pull/12)
* [J2735-2024 Part 2: SPAT, SSM & SRM](https://github.com/CDOT-CV/jpo-asn-pojos/pull/13)
* [J2735-2024 Part 3: MAP, PSM, RTCM](https://github.com/CDOT-CV/jpo-asn-pojos/pull/15)
* [J2735-2024 Part 4: BSM, TIM](https://github.com/CDOT-CV/jpo-asn-pojos/pull/16)
* [J2735-2024 Part 5: RSM, SDSM, CSR, EVA, RSA, ICA, NMEA, PVD, CCM](https://github.com/CDOT-CV/jpo-asn-pojos/pull/17)
* [J2735-2024 Part 6: RWM, PDM, PDR, PDC, MSCM, RUCCM, TAM](https://github.com/CDOT-CV/jpo-asn-pojos/pull/18)
* [J2735-2024 Part 7: RUCRM, TUM, TUMack, TLSM, Test & future use messages](https://github.com/CDOT-CV/jpo-asn-pojos/pull/19)
* [J2735-2024 Part 8: MessageFrames](https://github.com/CDOT-CV/jpo-asn-pojos/pull/20)
* [J2735-2024 Part 9: Test Data Tools](https://github.com/CDOT-CV/jpo-asn-pojos/pull/21)

This repository is intended to be pulled into the ODE as a submodule.

mcook42 committed:
refactor: use AbstractList to reduce boilerplate code in Asn1SequenceOf

iyourshaw committed:
Merge pull request #14 from CDOT-CV/mcook42/j2735-2024-type-refactor

Base Type Refactors

iyourshaw committed:
Pull in regenerated

iyourshaw committed:
Restore default constructors

iyourshaw committed:
Remove jar

iyourshaw committed:
EnumeratedDeserializr: throw appropriate exception instead of return null for invalid enum value, w/unit tests

iyourshaw committed:
Enum deserializer: return null if value actually is null or empty

iyourshaw committed:
Nested serializers/deserializers for open types and booleans.  Rawtypes annotations

iyourshaw committed:
MapData classes

iyourshaw committed:
PSM classes

iyourshaw committed:
RTCM

iyourshaw committed:
BSM classes

iyourshaw committed:
Revert "BSM classes"

This reverts commit 0b7f62a9d69550073b71f12d984b96738f2529e1.

iyourshaw committed:
Include tests

iyourshaw committed:
Bitstring test base class.  Remove rtcm message frame.

iyourshaw committed:
update script

iyourshaw committed:
BSMs

iyourshaw committed:
TIMs

iyourshaw committed:
RSM, SDSM

iyourshaw committed:
CSR

iyourshaw committed:
EVA, RSA

iyourshaw committed:
IC, NMEA

iyourshaw committed:
RSA test

iyourshaw committed:
PVD

iyourshaw committed:
CCM

iyourshaw committed:
RWM, PDM, PDR

iyourshaw committed:
PDC

iyourshaw committed:
runtime updates from main branch

iyourshaw committed:
Merge branch 'j2735-2024-part-3' into j2735-2024-part-4

iyourshaw committed:
Merge branch 'j2735-2024-part-4' into j2735-2024-part-5

iyourshaw committed:
Merge branch 'j2735-2024-part-5' into j2735-2024-part-6

iyourshaw committed:
Merge branch 'j2735-2024-part-6' of https://github.com/CDOT-CV/jpo-asn-pojos into j2735-2024-part-6

iyourshaw committed:
MSCM

iyourshaw committed:
RUCCM, TAM

iyourshaw committed:
RUCRM, TUM, TUMack

iyourshaw committed:
TLSM

iyourshaw committed:
Test and null messages

iyourshaw committed:
Rest of Common

iyourshaw committed:
MessageFrames and remaining classes

iyourshaw committed:
Remaining tests

iyourshaw committed:
Test generator

iyourshaw committed:
TAM tests

iyourshaw committed:
Merge branch 'j2735-2024-part-6' into j2735-2024-part-7

iyourshaw committed:
Remove print statements from test

iyourshaw committed:
Merge from part 7 and resolve

iyourshaw committed:
Fix test logging

iyourshaw committed:
Merge branch 'j2735-2024-part-8' into j2735-2024-part-9

iyourshaw committed:
Update testgen instructions

iyourshaw committed:
Remove update.sh

payneBrandon committed:
Merge pull request #13 from CDOT-CV/j2735-2024-part-2

J2735-2024 Part 2: SPAT, SSM & SRM

iyourshaw committed:
Add documentation to update.sh

iyourshaw committed:
Resolve conflict

iyourshaw committed:
Remove unused imports

iyourshaw committed:
Revert open type/RootSequenceOfSerializer

iyourshaw committed:
Merge branch 'j2735-2024-part-3' into j2735-2024-part-4

iyourshaw committed:
Merge branch 'j2735-2024-part-4' into j2735-2024-part-5

iyourshaw committed:
Add regression test for octet string validation bug

mcook42 committed:
refactor: delegate from default constructor in Asn1OctetString to prevent drift

mcook42 committed:
refactor: Asn1OctetString for better validation and clarity

Enhanced validation logic for hex strings and byte arrays with size constraints. Improved method documentation and added null checks to ensure robustness. Introduced utility methods for consistent length verification.

mcook42 committed:
chore: move Asn1OctetString constructors to top of file

mcook42 committed:
refactor: Asn1Bitstring constructors and improve exceptions

Added detailed Javadoc comments for constructors to enhance clarity. Introduced an overloaded constructor for consistent size handling. Improved exception messages to provide more context when input strings are too short.

mcook42 committed:
chore: update exception message to represent discrepancy between sizes

iyourshaw committed:
Add some notes to README

iyourshaw committed:
edit readme

iyourshaw committed:
edit readme

iyourshaw committed:
edit readme

iyourshaw committed:
Merge branch 'j2735-2024-part-3' into mcook42/suggest/j2735-2024-part-3

iyourshaw committed:
Add tests for deserializing octet string xml with various types of whitespace

mcook42 committed:
Revert "Add tests for deserializing octet string xml with various types of whitespace"

This reverts commit 28a2dbcdfebf73d441223e5109415415b924ea59.

mcook42 committed:
chore: trim hexString before validating size is multiple of 2

iyourshaw committed:
Merge pull request #22 from CDOT-CV/mcook42/suggest/j2735-2024-part-3

Suggested updates found while reviewing Part 3

payneBrandon committed:
Merge pull request #15 from CDOT-CV/j2735-2024-part-3

J2735-2024 Part 3 of 9: MAP, PSM, RTCM

iyourshaw committed:
Merge branch 'develop' into j2735-2024-part-4

iyourshaw committed:
VehicleAxlesAndWeight

iyourshaw committed:
Remove message frame tests.  Add VehiclAxlesAndWeightInfo

iyourshaw committed:
VehicleEventFlags test

iyourshaw committed:
TemporaryIDPointer

iyourshaw committed:
Bitstring fixes for variable length, non-named

iyourshaw committed:
Merge branch 'j2735-2024-part-4' into j2735-2024-part-5

iyourshaw committed:
Merge from part 5

iyourshaw committed:
RootSequenceOfSerializer

iyourshaw committed:
Merge from part 6

iyourshaw committed:
Merge branch 'j2735-2024-part-7' into j2735-2024-part-8

iyourshaw committed:
OpenType serdes

iyourshaw committed:
Merge from part 8

mcook42 committed:
refactor: use IntegerDeserializer directly instead of defining multiple implementations

mcook42 committed:
refactor: delete custom IntegerDeserializer.java in favor of JsonCreator annotations

iyourshaw committed:
Remove deserializers from all integers

iyourshaw committed:
Test deserializing large xml integer

iyourshaw committed:
Merge pull request #23 from CDOT-CV/mcook42/suggest/j2735-2024-part-4

Remove custom IntegerDeserializer

iyourshaw committed:
add hex tests to HeadingSliceTest

iyourshaw committed:
bitstring serdes: allow configurable human readable json

iyourshaw committed:
Create HumanReadableJsonBitstrings module

iyourshaw committed:
Tests

iyourshaw committed:
Javadoc

iyourshaw committed:
Note in runtime README

iyourshaw committed:
Test for ODE-style json TIM

payneBrandon committed:
Merge pull request #16 from CDOT-CV/j2735-2024-part-4

J2735-2024 Part 4 of 9: BSM, TIM

iyourshaw committed:
Merge branch 'develop' into j2735-2024-part-5

iyourshaw committed:
Update script, pull regenerated integers

iyourshaw committed:
Merge branch 'j2735-2024-part-5' into j2735-2024-part-6

iyourshaw committed:
Merge runtime updates

iyourshaw committed:
Handle deserializing JER format with bitstring value/length object

iyourshaw committed:
Bitstring serializer fix: serialize with variable format

iyourshaw committed:
Update MSCM json test

iyourshaw committed:
Updates to cli tool from mouse branch of asn1c

iyourshaw committed:
gitattributes for binary files

iyourshaw committed:
Add rawtypes annotations

iyourshaw committed:
suppress raw types warning on all ParameterizedTypeDeserializer implementations

iyourshaw committed:
Merge branch 'j2735-2024-part-9' into j2735-2024-part-9-var-bitstring-encoding

iyourshaw committed:
New jer converter.  Update tests with variable and extensible bitstrings for bsm, ic, sdsm, pvd.  Fix bitstring logic

iyourshaw committed:
Regenerate bitstrings with corrected hasExtensionMarker

iyourshaw committed:
Update tests for messages with extensible bitstrings

iyourshaw committed:
Add documentation for RootSeqeunceOfSerializer.  Fix TollPointMap.

iyourshaw committed:
Regenerate

iyourshaw committed:
Update script.  Pull updates from part 9

iyourshaw committed:
Update runtime and tests for variable length bitstrings

iyourshaw committed:
Add comments on scope of the standard and links to readme

iyourshaw committed:
typo

payneBrandon committed:
Merge pull request #17 from CDOT-CV/j2735-2024-part-5

J2735-2024 Part 5 of 9: RSM, SDSM, CSR, EVA, RSA, ICA, NMEA, PVD, CCM

iyourshaw committed:
Merge branch 'develop' into j2735-2024-part-6

iyourshaw committed:
Remove stray .java file from test resources dir

iyourshaw committed:
Remove unused import

iyourshaw committed:
Merge pull request #18 from CDOT-CV/j2735-2024-part-6

J2735-2024 Part 6 of 9: RWM, PDM, PDR, PDC, MSCM, RUCCM, TAM

iyourshaw committed:
Merge branch 'develop' into j2735-2024-part-7

iyourshaw committed:
Updates from part 9

iyourshaw committed:
Merge from develop

iyourshaw committed:
Fix tlsm test

iyourshaw committed:
roundtrip script

iyourshaw committed:
Fix TLSM test

iyourshaw committed:
Remove unused

iyourshaw committed:
Null type serdes

iyourshaw committed:
null tests

iyourshaw committed:
NullDeserializer is working, psm2 test

iyourshaw committed:
message frame test

iyourshaw committed:
RptRoadwayEventsTest

iyourshaw committed:
RptRoadwayEventsTests

iyourshaw committed:
Regenerate null types with deserializers

iyourshaw committed:
Tests for null types

iyourshaw committed:
MessageFrame tests

iyourshaw committed:
Tests for null choice types

iyourshaw committed:
Regenerate Asn1Null types

iyourshaw committed:
Tests for null choice types

iyourshaw committed:
Remove unused

iyourshaw committed:
Update jpo-asn-runtime/src/main/java/us/dot/its/jpo/asn/runtime/serialization/NullDeserializer.java

Co-authored-by: Matt Cook <mattheworion.cook@gmail.com>

iyourshaw committed:
Merge branch 'j2735-2024-part-7' into j2735-2024-part-8

iyourshaw committed:
Merge in updates from 7 and 9

iyourshaw committed:
named integers

iyourshaw committed:
Generate named integers

iyourshaw committed:
itis code tests

iyourshaw committed:
Coverage

iyourshaw committed:
Named integer test coverage

iyourshaw committed:
Remove unused

iyourshaw committed:
Named integer values with tests

iyourshaw committed:
Named integer values from value assignments

iyourshaw committed:
Test for DSRCmsgID

iyourshaw committed:
Integer values

iyourshaw committed:
Remove unused

iyourshaw committed:
Remove unused

iyourshaw committed:
Generator: Use spotless plugin to apply Google code format

iyourshaw committed:
Adding unit tests for test generator cli

iyourshaw committed:
Fix nested SEQUENCE types in PVD.VehicleStatus

iyourshaw committed:
Add ProbeVehicleData tests

iyourshaw committed:
Tests for generating all message types

iyourshaw committed:
Fix ints without constraints

iyourshaw committed:
Fixes from part 9

iyourshaw committed:
pvd tests

iyourshaw committed:
merge from part 8

iyourshaw committed:
Test generator documentation

iyourshaw committed:
Roundtrip script docs

iyourshaw committed:
Code cleanup

iyourshaw committed:
Add examples to docs. Optional flags

iyourshaw committed:
test log config

iyourshaw committed:
usage/readme

iyourshaw committed:
Remove

iyourshaw committed:
Merge branch 'j2735-2024-part-8' into j2735-2024-part-9

iyourshaw committed:
edit

iyourshaw committed:
ignore IDE files

iyourshaw committed:
ignore IDE files

iyourshaw committed:
Remove stray map file

iyourshaw committed:
Merge branch 'j2735-2024-part-8' into j2735-2024-part-9

iyourshaw committed:
Add test-generator coverage reports to CI.  Make the coverage comment 'update' instead of create a new comment every time.

iyourshaw committed:
Fix ci duplicate name

iyourshaw committed:
Separate task for each coverage report

iyourshaw committed:
coverate reports

iyourshaw committed:
typo in readme

iyourshaw committed:
typos in readme

iyourshaw committed:
rawtypes annotations

iyourshaw committed:
rawtypes annotations

iyourshaw committed:
Merge branch 'j2735-2024-part-7' into j2735-2024-part-8

iyourshaw committed:
Fix Maven test instructions and coverage aggregation

iyourshaw committed:
Node on vscode setting

iyourshaw committed:
Merge branch 'j2735-2024-part-7' into j2735-2024-part-8

iyourshaw committed:
Remove all exclusions from maven test coverage

iyourshaw committed:
Merge branch 'j2735-2024-part-8' into j2735-2024-part-9

payneBrandon committed:
Merge pull request #19 from CDOT-CV/j2735-2024-part-7

J2735-2024 Part 7 of 9: RUCRM, TUM, TUMack, TLSM, Test & future use messages

iyourshaw committed:
Merge branch 'develop' into j2735-2024-part-8

iyourshaw committed:
Merge branch 'develop' into j2735-2024-part-9

iyourshaw committed:
delete empty files

mcook42 committed:
refactor: encapsulate Asn1Integer.value property

iyourshaw committed:
Use jacoco aggregation plugin

iyourshaw committed:
Change whitespace to test runtime coverage

iyourshaw committed:
update docs

iyourshaw committed:
Fix artifacts upload

iyourshaw committed:
comment label

iyourshaw committed:
Merge pull request #25 from CDOT-CV/j2735-2024-part-9-coverage-aggregation

Test using jacoco coverage aggregation plugin

iyourshaw committed:
Add runtime to task names

iyourshaw committed:
Add runtime to task names

iyourshaw committed:
Fix

iyourshaw committed:
Remove unused import

iyourshaw committed:
Merge branch 'j2735-2024-part-8' into j2735-2024-part-9

mcook42 committed:
Update jpo-asn-j2735-2024/src/main/java/us/dot/its/jpo/asn/j2735/r2024/RoadWeatherMessage/NTCIPEssPavementTreatmentLocation.java

Co-authored-by: Ivan Yourshaw <39739503+iyourshaw@users.noreply.github.com>

mcook42 committed:
Update jpo-asn-j2735-2024/src/main/java/us/dot/its/jpo/asn/j2735/r2024/TravelerInformation/ITIStextPhrase.java

Co-authored-by: Ivan Yourshaw <39739503+iyourshaw@users.noreply.github.com>

mcook42 committed:
chore: use update-index --chmod=+x to track all gradlew scripts as executable

mcook42 committed:
refactor: update value assignments to use setValue method

Replaced direct value assignments with the setValue method across multiple classes for better encapsulation and consistency. Also updated access modifiers in Asn1CharacterString to enforce proper encapsulation.

mcook42 committed:
refactor: encapsulate fields

Changed protected and package-private fields to private and provided getter methods to enhance encapsulation and maintainability. Updated references to these fields to use the new getter methods, ensuring consistency across the codebase.

iyourshaw committed:
Merge pull request #24 from CDOT-CV/mcook42/suggest/j2735-2024-part-8

refactor: encapsulate Asn1Integer.value property

iyourshaw committed:
Merge branch 'j2735-2024-part-8' into j2735-2024-part-9

iyourshaw committed:
update testgen jar

mcook42 committed:
Merge branch 'j2735-2024-part-8' of github.com:CDOT-CV/jpo-asn-pojos into mcook42/suggest/j2735-2024-part-8-part2

mcook42 committed:
refactor: Asn1RelativeOID and RelativeRoadAuthorityID

Refactored Asn1RelativeOID to remove the no-arg constructor and make the value field final, improving immutability. Updated RelativeRoadAuthorityID to include a JsonCreator annotation while removing the no-arg constructor for consistency. These changes enhance object safety and simplify usage.

mcook42 committed:
fix: correct constructor parameter in BooleanDeserializer

Updated the constructor to accept the provided value type instead of a hardcoded class. Also removed an unnecessary suppression warning to improve code clarity.

mcook42 committed:
refactor: change constructors to protected visibility

Updated constructors for Asn1Bitstring and Asn1CharacterString to use protected visibility instead of public. This change limits direct instantiation outside the package, ensuring better encapsulation and supporting inheritance where needed.

mcook42 committed:
refactor: simplify Boolean deserializers and remove redundancies

Removed custom inner deserializer classes and replaced them with a generalized `BooleanDeserializer` implementation. This reduces code duplication across multiple classes and improves maintainability by centralizing deserialization logic.

mcook42 committed:
refactor: centralize BitString deserialization logic

Removed custom deserializers from individual classes and centralized the logic into a unified `BitStringDeserializer`. This improves maintainability and simplifies the codebase by handling deserialization contextually in a single location.

mcook42 committed:
fix: handle null BeanProperty in BooleanDeserializer

Ensure proper handling of null BeanProperty in the `createContextual` method. This fix prevents potential null pointer exceptions by using the contextual type when the property is not provided.

mcook42 committed:
Merge branch 'j2735-2024-part-9' of github.com:CDOT-CV/jpo-asn-pojos into mcook42/suggest/j2735-2024-part-8-part2

mcook42 committed:
refactor: change populateRandom to return the instance

Modified all populateRandom methods to return the given instance, ensuring consistency across all generator classes. This allows us to respect encapsulation

payneBrandon committed:
Merge pull request #20 from CDOT-CV/j2735-2024-part-8

J2735-2024 Part 8 of 9: MessageFrames

iyourshaw committed:
Merge branch 'develop' into j2735-2024-part-9

iyourshaw committed:
Update example

iyourshaw committed:
Use Jackson exceptions, remove hardcoded example

iyourshaw committed:
spotless plugin/google format

iyourshaw committed:
spotlessApply google format

iyourshaw committed:
Revert format changes to generator

iyourshaw committed:
Fix pojo spotless config.  Apply spotless to generator files

mcook42 committed:
Merge pull request #27 from CDOT-CV/iyourshaw/suggest/j2735-2024-part-8-part2

Suggestions for "Suggestion: Refactor Deserializers"

iyourshaw committed:
Merge pull request #26 from CDOT-CV/mcook42/suggest/j2735-2024-part-8-part2

Suggestion: Refactor Deserializers

iyourshaw committed:
Update jpo-asn-j2735-2024/README.md

Co-authored-by: Matt Cook <mattheworion.cook@gmail.com>

iyourshaw committed:
converter-new-jer example

iyourshaw committed:
Merge branch 'j2735-2024-part-9' of https://github.com/CDOT-CV/jpo-asn-pojos into j2735-2024-part-9

iyourshaw committed:
Document minified & pretty-printed examples

iyourshaw committed:
Document minified & pretty-printed examples

payneBrandon committed:
Merge pull request #21 from CDOT-CV/j2735-2024-part-9

J2735-2024 Part 9 of 9: Test Data Tools

iyourshaw committed:
add release notes

iyourshaw committed:
correct semantic versioning: change to '1.0.0'

iyourshaw committed:
Move release notes to docs folder

dan-du-car commented:
Unfortunately, I am unable to comment on each individual file as it is too big and my browser is frozen when I open `File Changes`. Below are a few questions/comments:

- Inside .github/workflows folder, there is a test.yml file. Could we be consistent on the yaml file naming practice? We usually use ci.yml for build, run unit test and generate code coverage processes. Note: Add coverage as PR comment is a really good feature, and we are looking into leverage this in our PRs as well. Thanks!
- In the appropriate README.md files, could we add a section how the J2735 2024 specification to java class source files in folder `jpo-asn-j2735-2024` are generated? Which tool is used? Are the java source files inside folder `jpo-asn-runtime` generated as well? and how?
- Shall gradlew commands to be executable? It can be pushed with executable permission.
![image](https://github.com/user-attachments/assets/f6d279ee-16c0-4232-b94a-e3b31ff3ca1a)
- Is this project sole purpose to provide java tool/utility to generate J2735 2024 test data? It provides awesome serialization/deserialize for J2735 conformance ASN.1 message. Would it also be intended to be used by other projects that require internal  serialization/deserialize of J2735 messages?

iyourshaw commented:
> * Is this project sole purpose to provide java tool/utility to generate J2735 2024 test data? It provides awesome serialization/deserialize for J2735 conformance ASN.1 message. Would it also be intended to be used by other projects that require internal  serialization/deserialize of J2735 messages?

The immediate and main purpose of this is to import these POJOs into the `jpo-ode` project to replace the existing hand-written classes there with a more complete and consistent implementation.  Currently the ODE has an earlier version of them for TIMs only (see [jpo-ode-plugins/travelerinformation](https://github.com/usdot-jpo-ode/jpo-ode/tree/develop/jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/travelerinformation)), but the idea is to enable refactoring all the message types along the same lines, plus to be able to support new message types more easily (eg SDSM and RSM).  The test generator exists as a side effect of creating and testing the POJOs, but we included it since we thought it could be useful to share as well.

For now we plan to use this repo as a git submodule in the ode, but eventually would like to make it available as a library in Maven central, for any other projects that may want to use it.

iyourshaw committed:
Rename file test.yml -> ci.yml

iyourshaw committed:
Note on asn1jvm tool

iyourshaw committed:
Make testgen/gradlew files executable in git

iyourshaw commented:
> * Inside .github/workflows folder, there is a test.yml file. Could we be consistent on the yaml file naming practice? We usually use ci.yml for build, run unit test and generate code coverage processes. Note: Add coverage as PR comment is a really good feature, and we are looking into leverage this in our PRs as well. Thanks!

Changed that file name.

> * In the appropriate README.md files, could we add a section how the J2735 2024 specification to java class source files in folder `jpo-asn-j2735-2024` are generated? Which tool is used? Are the java source files inside folder `jpo-asn-runtime` generated as well? and how?

The POJOs in `jpo-asn-j2735-2024` were generated using ASN1JVM which is an internal tool.
The java source files in `jpo-asn-runtime` were written manually, not generated.
Added a note here:
https://github.com/CDOT-CV/jpo-asn-pojos/blob/cdot-release_2025-q2/jpo-asn-j2735-2024/README.md#code-generation


> * Shall gradlew commands to be executable? It can be pushed with executable permission.

Good catch, I did "git update-index chmod=+x" on those..
